crates/bevy_render/src/color/mod.rs

```diff
@@ -326,26 +326,17 @@ impl Color {
 
     /// Get red in sRGB colorspace.
     pub fn r(&self) -> f32 {
-        match self.as_rgba() {
-            Color::Rgba { red, .. } => red,
-            _ => unreachable!(),
-        }
+        self.as_rgba_f32()[0]
```

Comment:
I (think) I like that this increases the utility of this function by doing the conversion when necessary. But this will also do unnecessary work for representations like RgbaLinear, which will call `linear_to_nonlinear_srgb` for r, g, and b. Theres also the question of moving more data around (one dimension versus all 4). Afaik, these unused converted values wont get optimized out because these functions aren't inlined (as they are public?).  And even if we can prove they will be inlined (and also prove that the unnecessary work will be optimized out), idk if its worth putting that extra work on the compiler, when compared to just inlining the logic we need. DRY is nice, but it isn't something to be pursued at all costs, especially for core types like Color.